pkgs/top-level/all-packages.nix

```diff
@@ -267,6 +267,10 @@ with pkgs;
     buildGoModule = buildGo123Module;
   };
 
+  butler = callPackage ../by-name/bu/butler/package.nix {
+    inherit (darwin.apple_sdk.frameworks) Cocoa;
+  };
+
```

Comment:
```suggestion
```

Reply:
removed the edits to all-packages.nix as suggested.

Reply:
I'm not sure how butler gets defined though, how is it intended to be added?

Reply:
It's automatically detected by the by-name mechanism. See pkgs/by-name/README.md

Reply:
Oooh!! That's very cool. Thank you for being patient with me. I've been reading lots but it's easy to get overwhelmed.